mods/cnc/maps/nod04a/nod04a.lua

```diff
@@ -0,0 +1,296 @@
+				Trigger.RemoveFootprintTrigger(id)
+			end
+		end
+	end)
+
+	Trigger.OnEnteredFootprint(Atk2CellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			MyActors = getActors(GDI, { ['jeep'] = 1 })
+			Utils.Do(MyActors, function(actor)
+				Atk2Movement(actor)
+			end)
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.OnEnteredFootprint(GcivCellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			MyActors = getActors(GDI, { ['e1'] = 2,['c1'] = 1,['c2'] = 1,['c3'] = 1,['c4'] = 1 })
+			Utils.Do(MyActors, function(actor)
+				GcivMovement(actor)
+			end)
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.AfterDelay(Atk1TriggerFunctionTime, Atk1TriggerFunction)
+
+	Trigger.OnEnteredFootprint(Atk4CellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			MyActors = getActors(GDI, { ['e1'] = 2,['e2'] = 1 })
+			Utils.Do(MyActors, function(actor)
+				Atk4Movement(actor)
+			end)
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.AfterDelay(XxxxTriggerFunctionTime, XxxxTriggerFunction)
+	Trigger.AfterDelay(YyyyTriggerFunctionTime, YyyyTriggerFunction)
+	Trigger.AfterDelay(ZzzzTriggerFunctionTime, ZzzzTriggerFunction)
+
+	Trigger.OnEnteredFootprint(DelxCellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			XxxxTriggerSwitch = true
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.OnEnteredFootprint(DelyCellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			YyyyTriggerSwitch = true
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.OnEnteredFootprint(DelzCellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			ZzzzTriggerSwitch = true
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.OnEnteredFootprint(ShowCivilianCellTriggerActivator, function(a, id)
+		Utils.Do(NodCivilians, function(civilian)
+			local Actor = Reinforcements.Reinforce(Nod, civilian[1], civilian[2], 15, nil)
+			NodCiviliansActors[#NodCiviliansActors+1] = Actor[1]
+		end)
+		NodObjective2 = Nod.AddPrimaryObjective("Protect the civilians that supports Nod.")
+		Trigger.OnAllKilled(NodCiviliansActors, function()
+			Nod.MarkFailedObjective(NodObjective2)
+		end)
+		Trigger.RemoveFootprintTrigger(id)
+	end)
+
+	Trigger.OnAllKilled(WinActorTriggerActivator, function()
+		Nod.MarkCompletedObjective(NodObjective1)
+		Nod.MarkCompletedObjective(NodObjective2)
+	end)
+	
+	GDIObjective = GDI.AddPrimaryObjective("Eliminate all Nod forces in the area.")
+	NodObjective1 = Nod.AddPrimaryObjective("Kill all civilian which support the GDI.")
+end
+
+Tick = function()
+	if Nod.HasNoRequiredUnits()  then
+		GDI.MarkCompletedObjective(GDIObjective)
+	end
+end
+
+OnAnyDamaged = function(actors, func)
+	Utils.Do(actors, function(actor)
+		Trigger.OnDamaged(actor, func)
+	end)
+end
+
+IdleHunt = function(unit) 
+	if not unit.IsDead then 
+		Trigger.OnIdle(unit, unit.Hunt) 
+	end 
+end
+
+getActors = function(owner, units)
```

Comment:
This should make use of the new Lua API function from https://github.com/OpenRA/OpenRA/pull/7463.